### DISCUSSION TITLE
Libre 2 EU Sensor Expiry reported incorrectly

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2250] [General]
The reported end time, on the main screen, does not agree with what is shown on the system status page.  And it changes with time.

![296716808_5211729188874259_2714584953830568599_n](https://user-images.githubusercontent.com/51497406/181909454-3b62612b-a8ed-4c33-b1ce-dce75b78c83b.jpg)

![296735094_5211689208878257_4559462783733941584_n](https://user-images.githubusercontent.com/51497406/181909338-deaa54b5-a39e-47eb-82ca-a06938384d55.jpg)

![296752061_5211722032208308_4462630514443318808_n](https://user-images.githubusercontent.com/51497406/181909362-61355db8-6534-4e96-ab85-d72bbb95c065.jpg)

**Comment by timkuu** [comment]
That is my sensor. The ending time seems to change constantly abt 2h 9mins ahead the phone time. The starting time was reported correctly in xDrip, and the System Status shows it correctly.

**Comment by jamorham** [comment]
Please describe which collection method is being used. Thanks

**Reply by timkuu** [reply]
EU Libre 2 with patched LibreLink.

**Comment by jamorham** [comment]
In the screenshot it shows 13.9 days I think (hard to read) Is that figure correct?

**Reply by timkuu** [reply]
Yes, 13.9d is right, but the age is incorrect. The true age at that point was 12d 19h as seen on the System Status screenshot. 
My sensor ended already, and I'll apply a new one this evening. I can report how the sensor age on the main window and System Status progress from the start onwards.

**Comment by timkuu** [comment]
After starting a new sensor last evening the sensor expiry displays correctly. 

The initial issue could have come up because I updated xDrip during an active sensor. My previous xDrip version did not have this feature. 

![Screenshot_20220802-085329](https://user-images.githubusercontent.com/64281537/182302097-983cb451-ec43-4d1d-add0-2a4b0da93752.png)

**Reply by jamorham** [reply]
Were you previously using a different type of Libre collector? I think that is where this bug would come from and I think I have addressed it in today's nightly

**Reply by timkuu** [reply]
No, LL patch only. 

Well, I do take the last juices out of the sensor with oop2 but that's only for 12 hours after the sensor has died. 

This issue occured with the patch after I updated xDrip from ver 20220713 to 20220726, when the sensor had 2 to 3 days of life left. I had been using the patch from the start.

With the presemt sensor the expiry time still reports correct.

**Reply by jamorham** [reply]
Yes it will be OOP2 followed by patched which triggers this condition. The latest nightly should resolve it.

**Reply by timkuu** [reply]
Thanks! Great job!
